subsys/bluetooth/host/settings.c

```diff
@@ -323,3 +297,179 @@ int bt_settings_init(void)
 
 	return 0;
 }
+
+int bt_settings_store(const char *key, uint8_t id, const bt_addr_le_t *addr, const void *value,
+		      size_t val_len)
+{
+	char id_str[4];
+	char key_str[BT_SETTINGS_KEY_MAX];
+
+	if (addr) {
+		if (id) {
+			u8_to_dec(id_str, sizeof(id_str), id);
+		}
+
+		bt_settings_encode_key(key_str, sizeof(key_str), key, addr, (id ? id_str : NULL));
+	} else {
+		if (strlen(key) > BT_SETTINGS_KEY_MAX - 4) {
+			return -EINVAL;
+		}
+
+		strcpy(&key_str[0], "bt/");
+		strcpy(&key_str[3], key);
```

Comment:
There's a risk that static analysers like Coverity might trip over the use of strcpy instead of its safe strncpy variant. However, using just `snprintk()` is likely the simplest here:
```c
snprintk(key_str, sizeof(key_str), "bt/%s", key);
```